File: candid/tests/assets/codegen/examples/linkedup.rs
pub type UserId_2 = UserId;
pub type UserId = principal;
#[derive(Clone)]
pub struct Profile_2 {
    pub id: UserId,
    pub imgUrl: String,
    pub title: String,
    pub education: String,
    pub experience: String,
    pub company: String,
    pub lastName: String,
    pub firstName: String,
}
pub type Profile = Profile_2;
#[derive(Clone)]
pub struct NewProfile_2 {
    pub imgUrl: String,
    pub title: String,
    pub education: String,
    pub experience: String,
    pub company: String,
    pub lastName: String,
    pub firstName: String,
}
pub type NewProfile = NewProfile_2;
pub trait Actor {
    fn connect(
        arg0: UserId_2,
    ) -> std::pin::Pin<std::boxed::Box<impl std::future::Future<Output = ()>>>;
    fn create(
        arg0: NewProfile,
    ) -> std::pin::Pin<std::boxed::Box<impl std::future::Future<Output = ()>>>;
    fn get(arg0: UserId_2) -> Profile;
    fn getConnections(
        arg0: UserId_2,
    ) -> std::pin::Pin<std::boxed::Box<impl std::future::Future<Output = Vec<Profile>>>>;
    fn getOwnId() -> UserId_2;
    fn healthcheck() -> std::pin::Pin<std::boxed::Box<impl std::future::Future<Output = bool>>>;
    fn isConnected(
        arg0: UserId_2,
    ) -> std::pin::Pin<std::boxed::Box<impl std::future::Future<Output = bool>>>;
    fn search(arg0: String) -> Vec<Profile>;
    fn update(
        arg0: Profile,
    ) -> std::pin::Pin<std::boxed::Box<impl std::future::Future<Output = ()>>>;
}

File: candid/tests/assets/codegen/examples/linkedup.js
export default function({ IDL }) {
  const UserId_2 = UserId;

  const UserId = IDL.Principal;

  const Profile_2 = IDL.Record({
    'id': UserId,
    'imgUrl': IDL.Text,
    'title': IDL.Text,
    'education': IDL.Text,
    'experience': IDL.Text,
    'company': IDL.Text,
    'lastName': IDL.Text,
    'firstName': IDL.Text,
  });

  const Profile = Profile_2;

  const NewProfile_2 = IDL.Record({
    'imgUrl': IDL.Text,
    'title': IDL.Text,
    'education': IDL.Text,
    'experience': IDL.Text,
    'company': IDL.Text,
    'lastName': IDL.Text,
    'firstName': IDL.Text,
  });

  const NewProfile = NewProfile_2;

  return IDL.Service({
    'connect': IDL.Func([UserId_2], [], []),
    'create': IDL.Func([NewProfile], [], []),
    'get': IDL.Func([UserId_2], [Profile], ['query']),
    'getConnections': IDL.Func([UserId_2], [IDL.Vec(Profile)], []),
    'getOwnId': IDL.Func([], [UserId_2], ['query']),
    'healthcheck': IDL.Func([], [IDL.Bool], []),
    'isConnected': IDL.Func([UserId_2], [IDL.Bool], []),
    'search': IDL.Func([IDL.Text], [IDL.Vec(Profile)], ['query']),
    'update': IDL.Func([Profile], [], []),
  });
}
